pkg/proxy/backend/authenticator.go

```diff
@@ -160,7 +161,7 @@ func (auth *Authenticator) handshakeFirstTime(logger *zap.Logger, clientIO *pnet
 	auth.attrs = resp.Attrs
 
 	// In case of testing, backendIO is passed manually that we don't want to bother with the routing logic.
-	backendIO, err := getBackendIO(auth, auth, resp)
+	backendIO, err := getBackendIO(auth, auth, resp, 5*time.Second)
```

Comment:
There is a constant, `DialTimeout` in `backend_conn.go`.

Reply:
It's different. The one passed to `getBackendIO` is the total timeout for all retries.
BTW, `DialTimeout` is also 5 seconds. Maybe it should be less?